dev/integration_tests/web_e2e_tests/test_driver/text_editing_integration.dart

```diff
@@ -156,23 +177,24 @@ void main() {
     await tester.tap(find.byKey(const Key('input')));
 
     // A native input element will be appended to the DOM.
-    final List<Node> nodeList = document.getElementsByTagName('input');
+    final List<Node> nodeList = findElements('input');
     expect(nodeList.length, equals(1));
-    final InputElement input =
-    document.getElementsByTagName('input')[0] as InputElement;
+    final InputElement input = nodeList[0] as InputElement;
 
     // Press and release CapsLock.
     dispatchKeyboardEvent(input, 'keydown', <String, dynamic>{
       'key': 'CapsLock',
       'code': 'CapsLock',
       'bubbles': true,
       'cancelable': true,
+      'composed': true,
```

Comment:
`composed` needs to be set to `true` in synthetic events (similar to `bubbles`) so the event propagates in a similar way as to when it's dispatched by the browser. [Docs](https://developer.mozilla.org/en-US/docs/Web/API/Event/composed).